src/matrix-tmp.h

```diff
@@ -7,15 +7,15 @@
 
 #include "matrix.h"
 
-template <Elementable Element>
+template <IsMatrixableElement Element>
```

Comment:
the name is not good becuse you ask a question " is element matrixable" but you use word say ability.